c/core.go

```diff
@@ -47,25 +47,36 @@ func WithShutdown(s Shutdown) Opt {
 // Depending on the `Shutdown` values used in the `ChildSpec` , if the start
 // function does not respect the given context, the parent supervisor will
 // either block forever or leak goroutines after a timeout has been reached.
-func New(name string, start func(context.Context) error, opts ...Opt) ChildSpec {
-	return NewWithNotifyStart(name, func(ctx context.Context, notifyChildStart func()) error {
-		notifyChildStart()
-		return start(ctx)
-	}, opts...)
+func New(name string, childMain func(context.Context) error, opts ...Opt) ChildSpec {
```

Comment:
> You might want to add some semantical context around `childMain()`

What are you thinking? a type alias?

> Start gave me the impression of an async operation. `childMain()` feels synchronous. Is that what you now intend?

No, I renamed it thinking that it might help to understand what is code provided by the client code. You are right about this code executing asynchronously. I want to keep this; should I rename this back to `start`?

Reply:
Renaming to `startChildMain`

Reply:
All I meant was a sentence about what `childMain()` is meant to be. If it is async, then yes, I would use the word `start`. In my mind (perhaps just me), `start` has an async connotation whereas `childMain()` feels more sync. Either way, just a sentence indicating that it is in fact _async_, would be fine, regardless of the name.